### PR TITLE
Bug 2087149: Add watch predicates to kubeconfig controller

### PR DESCRIPTION
**Pull request by Fedosin**
Now it doesn't react on changes in the managed secrets, which allows users to modify them.

### CODE REVIEW

File: pkg/controllers/kubeconfig/watch_predicates.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		UpdateFunc:  func(e event.UpdateEvent) bool { return isOwnedTokenSecret(e.ObjectNew) },
+		DeleteFunc:  func(e event.DeleteEvent) bool { return isOwnedTokenSecret(e.Object) },
+		GenericFunc: func(e event.GenericEvent) bool { return isOwnedTokenSecret(e.Object) },
+	}
+}
+
+func kubeconfigSecretPredicate() predicate.Funcs {
+	isKubeconfigSecret := func(obj runtime.Object) bool {
+		secret, ok := obj.(*corev1.Secret)
+		if !ok {
```
Comment (alexander-demichev) [inline pkg/controllers/kubeconfig/watch_predicates.go:43]: are you sure that we want the code to panic and not use the pattern we have here https://github.com/openshift/cluster-capi-operator/blob/main/pkg/controllers/secretsync/watch_predicates.go#L21?

Comment (Fedosin) [reply pkg/controllers/kubeconfig/watch_predicates.go:43]: We decided to panic in these cases. For instance: https://github.com/openshift/cluster-control-plane-machine-set-operator/blob/main/pkg/controllers/controlplanemachineset/watch_filters.go#L56-L59

Here's @JoelSpeed comment: https://github.com/openshift/cluster-control-plane-machine-set-operator/pull/28#discussion_r844859820

**Review by alexander-demichev — APPROVED** [review]
/approve
/lgtm

**Pull request merged**